### DISCUSSION TITLE
Error when trying to use useRichTextEditorContext with tiptap

### DISCUSSION THREAD

**Discussion by mroseneck** [mantinedev/mantine#3182] [Q&A] [answered]
### What package has an issue

@mantine/tiptap

### Describe the bug

When I try to get the current editor object via "useRichTextEditorContext" in an otherwise working @mantine/tiptap setup and the useEditor hook, I'm getting the following error:

`Uncaught Error: RichTextEditor was not found in tree`.

I'm using mantine in the context of a ClojureScript project here, so while it is possible that that is the reason, I doubt it, because everything else works perfectly-

### What version of @mantine/hooks page do you have in package.json?

5.9.2

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by rtivital** [comment, marked as answer]
You are calling the hook outside of RichTextEditor component context. This hook can only be used in components that are rendered inside RichTextEditor component.

**Reply by mroseneck** [reply]
Thanks for the quick reply!

I'm basically trying to replicate the example here: https://mantine.dev/others/tiptap/#custom-controls
So the call definitely happens in the context of a "RichTextEditor" element.

Hm okay, I will try to figure it out then. The reason why I think I need the context is that I want to get the content of the editor as JSON data. Do you happen to know if there is another way to do that?

Regards

**Reply by rtivital** [reply]
Send a sandbox that reproduces the issue, template – https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6

**Reply by mroseneck** [reply]
Ah, thanks, when trying to reproduce the problem I saw my mistake. It was the way I was invoking the function.
